api/v1/qdrantcluster_types.go

```diff
@@ -693,17 +703,22 @@ type QdrantClusterStatus struct {
 	// +optional
 	AvailableNodeIndexes []int `json:"availableNodeIndexes,omitempty"`
 	// BootstrapNode specifies the node in the cluster which will be used for bootstrapping a new node.
-	// Should be 0...n-1 (default = 0)
+	// Should be a value from AvailableNodeIndexes.
+	// As default the value from AvailableNodeIndexes[0] will be used.
 	// +optional
 	BootstrapNode int `json:"bootstrapNode,omitempty"`
 	// If set the operator will scale down 1 peer and reset the bool.
 	// If you want to remove more then 1 peer, you need to repeat setting this bool
 	// +optional
 	ScaleDownAllowed bool `json:"scaleDownAllowed,omitempty"`
+	// The node index used in a scale down (see ScaleDownAllowed)
+	// If this field is not set the last index in AvailableNodeIndexes will be used.
+	ScaleDownNodeIndex *int `json:"ScaleDownNodeIndex,omitempty"`
 	// Conditions specifies the conditions of different checks on the cluster
 	// +optional
 	Conditions []metav1.Condition `json:"conditions,omitempty"`
 	// Operations tracks list of recent operation on the cluster. Operator uses this field to track the progress of an operation.
+	// In operator V2 this field is deprecated and Kubernetes events are used instead.
```

Comment:
Why deprecating this? We should be cautions with events. Each event generation is request sending to k8s api server, so in case of many events, we can create a significant load to k8s api server.. Event also have a TTL, so an observed might not be able to access them.

An operator should report to observers the status of the given resource through `.status`, possibly under `.status.conditions`

Reply:
This is already be the case, but wasn't documented. We know that there is a TTL, but for the regional clusters the events are added to Loki and will be preserved. The issue with the this operations field in V1 was that it became a very long (and large) field.

Reply:
How about private clusters? We should have a way to report all important statuses within resource status field.

> The issue with the this operations field in V1 was that it became a very long (and large) field.

What do you mean? What is the impact on the system?

I am fine if we replace this field with appropriate conditions.

Reply:
It is, see: Conditions

Reply:
[PS: Operations is used by the Python kopf framework to track status, we decoded to not port that portion]